tests/lib/common.py

```diff
@@ -56,11 +56,14 @@ def wait_for_result(func, *args, matcher=simple_matcher(True), attempts=20,
     """
 
     for i in range(attempts):
-        out = func(*args, **kwargs)
-        if decode:
-            out = decode(out)
-        if matcher(out):
-            return out
+        try:
+            out = func(*args, **kwargs)
+            if decode:
+                out = decode(out)
+            if matcher(out):
+                return out
+        except Exception:
+            pass
```

Comment:
I think we want to know the exception in case we reached the maximum retries.

Reply:
I added the code for that.